.github/workflows/ci.yml

```diff
@@ -196,6 +196,9 @@ jobs:
           export PATH=${HOME}/Library/Python/3.8/bin:${HOME}/Library/Python/3.9/bin:${HOME}/.local/bin:${PATH}
           if [ "$ASAN" == "true" ]; then
             export CFLAGS="-DASAN=1 ${CFLAGS}"
+            if [ "$CC" == "clang" ]; then
+              export LD_LIBRARY_PATH=$(dirname $(clang -print-file-name=libclang_rt.asan-x86_64.so)):${LD_LIBRARY_PATH}
+            fi
```

Comment:
IIRC for Clang (based off experience from #1910):

1. `-shared-libasan` is used to indicate that the ASAN library is a shared library. Every piece of ASAN-compiled code must tell its compiler and linker of this -- if even one isn't aware, the static version of ASAN will get linked in and things will not end well (i.e. you get `Your application is linked against incompatible ASan runtimes`).
2. As per https://github.com/systemd/systemd/blob/66a0e222937e145a1722640fab413608f565cf33/docs/TESTING_WITH_SANITIZERS.md#clang, this shared library is found using `LD_LIBRARY_PATH` by the dynamic linker (not sure why you're setting `LD_LIBRARY_PATH` here).
3. If you don't use `-shared-libasan`, then you cannot use `LD_PRELOAD` and this would XX some tests.

Reply:
> (not sure why you're setting `LD_LIBRARY_PATH` here).

Oh it's for the sdb executable; I truly forgot about this. Now I think you have to use `readelf` and examine carefully the dynamic symbols of `sdb_native` and its supporting Rizin shared libs --- all ASAN/UBSAN symbols must be UND.